compiler/AST/build.cpp

```diff
@@ -378,11 +396,63 @@ static BlockStmt* buildUseList(BaseAST* module, BlockStmt* list) {
   }
 }
 
+//
+// Given a string literal argument from a 'use' statement, process
+// that argument.  We assume it's either a "-llib" flag,
+// or a filename like "foo.h", "foo.c", "foo.o", etc.  
+//
+// - For the former, pass to addLibInfo(), the same function that
+//   handles command line -l flags.
+//
+// - Otherwise, assume it's the latter and pass it to our input file
+//   handler (which itself handles cases it doesn't recognize).
+//
+static void processUseOfString(const char* str) {
+  // If the string starts with -l, treat it as a library specifier
+  //
+  if (strncmp(str, "-l", 2) == 0) {
+    addLibInfo(str);
+  } else {
+    //
+    // otherwise, treat it as an input file
+    //
+    addSourceFile(str);
+  }
+}
+
 
-BlockStmt* buildUseStmt(CallExpr* modules) {
+//
+// Build a 'use' statement
+//
+BlockStmt* buildUseStmt(CallExpr* args) {
   BlockStmt* list = NULL;
-  for_actuals(expr, modules)
-    list = buildUseList(expr->remove(), list);
+
+  //
+  // Iterate over the expressions being used, processing them
+  //
+  for_actuals(expr, args) {
+    Expr* useArg = expr->remove();
+
+    //
+    // if this is a string argument to 'use', process it
+    //
+    if (const char* str = toImmediateString(useArg)) {
+      processUseOfString(str);
```

Comment:
`processUseOfString` is a confusing name to me. It's not so bad at the definition of the function since the comment is right there, but it's confusing here at the call site. Maybe `processStringInUseStmt`?